### PR TITLE
Habana 1.2.0 release update

### PR DESCRIPTION
**Pull request by omrialmog**
* Update docker image path to 1.1.1 release docker
* Added 1.9.1 pytorch
* Added 2.7.0 tensorflow

*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: habana/pytorch/buildspec.yml [hunk 1]
```diff
@@ -47,15 +47,15 @@ images:
         VAR_IN_DOCKERFILE: PT_TRAINING_URL
```
Comment (omrialmog) [inline habana/pytorch/buildspec.yml:47]: @waytrue17 Can you please update URI above to aws 1.9.1 + habana pytorch framework

File: habana/tensorflow/training/docker/2.7/py3/synapseai1.1.1/Dockerfile.hpu [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+ENV PYTHONUNBUFFERED=1
+ENV PYTHONIOENCODING=UTF-8
+ENV LANG=C.UTF-8
+ENV LC_ALL=C.UTF-8
+
+ARG PYTHON=python3.7
+ARG PYTHON_PIP=python3-pip
+ARG PIP=pip3.7
+
+ARG TF_URL=https://aws-tensorflow-binaries.s3.us-west-2.amazonaws.com/tensorflow/r2.5_aws/20210715_063140/nomkl/cpu/py37/tensorflow_cpu-2.5.0-cp37-cp37m-manylinux2010_x86_64.whl
```
Comment (omrialmog) [inline habana/tensorflow/training/docker/2.7/py3/synapseai1.1.1/Dockerfile.hpu:28]: @waytrue17 Can you please update to aws 2.7.0 tf framework

File: habana/pytorch/training/docker/1.9/py3/synapseai1.1.1/Dockerfile.hpu [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+RUN HOME_DIR=/root \
+ && curl -o ${HOME_DIR}/oss_compliance.zip https://aws-dlinfra-utilities.s3.amazonaws.com/oss_compliance.zip \
+ && unzip ${HOME_DIR}/oss_compliance.zip -d ${HOME_DIR}/ \
+ && cp ${HOME_DIR}/oss_compliance/test/testOSSCompliance /usr/local/bin/testOSSCompliance \
+ && chmod +x /usr/local/bin/testOSSCompliance \
+ && chmod +x ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh \
+ && ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh ${HOME_DIR} ${PYTHON} \
+ && rm -rf ${HOME_DIR}/oss_compliance*
+
+RUN curl -o /license.txt https://aws-dlc-licenses.s3.amazonaws.com/pytorch-1.7/license.txt
```
Comment (omrialmog) [inline habana/pytorch/training/docker/1.9/py3/synapseai1.1.1/Dockerfile.hpu:91]: @waytrue17 Can you please update to aws 1.9.1+habana pt framework

**Review by omrialmog — COMMENTED** [review]
@tejaschumbalkar will you be updating available_images.md when this images are uploaded?

File: habana/pytorch/buildspec.yml (hunk 1 above)
Comment (waytrue17) [reply habana/pytorch/buildspec.yml:47]: Thank you for the pointers. I'll be updating them when framework binaries are ready @omrialmog

**Comment by waytrue17** [comment]
Updated framework binaries for PT1.9.1 and TF2.7.0

File: habana/pytorch/buildspec.yml (hunk 1 above)
Comment (tejaschumbalkar) [reply habana/pytorch/buildspec.yml:47]: This configuration for specifying the binary is not needed. We can directly specify the binary in the dockerfile like [PT1.9 dockerfile](https://github.com/aws/deep-learning-containers/blob/master/pytorch/training/docker/1.9/py3/Dockerfile.cpu#L31)

Lets remove the `pytorch_binary` configuration from this file

File: habana/pytorch/buildspec.yml [hunk 4]
```diff
@@ -27,44 +27,44 @@ context:
...
-    os_version: &OS_VERSION ubuntu18.04
+    tag_python_version: &TAG_PYTHON_VERSION py38
+    synapseai_version: &SYNAPSEAI_VERSION synapseai1.1.1
+    os_version: &OS_VERSION ubuntu20.04
     tag: !join [ *VERSION, "-", *DEVICE_TYPE, "-", *TAG_PYTHON_VERSION, "-", *SYNAPSEAI_VERSION,
       "-", *OS_VERSION, "-example" ]
     docker_file: !join [ docker/, *SHORT_VERSION, /, *DOCKER_PYTHON_VERSION, /example,
       /Dockerfile., *DEVICE_TYPE ]
     download_artifacts:
       pytorch_binary:
```
Comment (tejaschumbalkar) [inline habana/pytorch/buildspec.yml:65]: Same here with regards to `pytorch_binary` configuration

File: habana/pytorch/training/docker/1.9/py3/synapseai1.1.1/Dockerfile.hpu [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    wget \
+    vim \
+    zlib1g-dev \
+    # Install dependent library for OpenCV
+    libgtk2.0-dev \
+ && rm -rf /var/lib/apt/lists/* \
+ && apt-get clean
+
+
+ARG PT_TRAINING_URL
```
Comment (tejaschumbalkar) [inline habana/pytorch/training/docker/1.9/py3/synapseai1.1.1/Dockerfile.hpu:34]: Lets remove this reference and directly use the PT binary

File: habana/pytorch/training/docker/1.9/py3/synapseai1.1.1/Dockerfile.hpu [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+COPY ${PT_TRAINING_URL} .
```
Comment (tejaschumbalkar) [inline habana/pytorch/training/docker/1.9/py3/synapseai1.1.1/Dockerfile.hpu:35]: This will also be not needed in that case

File: habana/pytorch/training/docker/1.9/py3/synapseai1.1.1/Dockerfile.hpu [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    psutil \
+    Pillow \
+    scipy \
+    pybind11 \
+    mpi4py==3.0.3 \
+    cmake==3.18.2.post1 \
+    torchnet \
+    "cryptography>3.2" \
+    werkzeug \
+ && pip uninstall -y dataclasses typing \
```
Comment (tejaschumbalkar) [inline habana/pytorch/training/docker/1.9/py3/synapseai1.1.1/Dockerfile.hpu:58]: @omrialmog Is this required? Can we remove this?

File: habana/pytorch/training/docker/1.9/py3/synapseai1.1.1/Dockerfile.hpu (hunk 3 above)
Comment (tejaschumbalkar) [reply habana/pytorch/training/docker/1.9/py3/synapseai1.1.1/Dockerfile.hpu:91]: @waytrue17 Lets update the license to `https://aws-dlc-licenses.s3.amazonaws.com/pytorch-1.9/license.txt`

File: habana/tensorflow/training/docker/2.7/py3/synapseai1.1.1/Dockerfile.hpu [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+RUN HOME_DIR=/root \
+   && curl -o ${HOME_DIR}/oss_compliance.zip https://aws-dlinfra-utilities.s3.amazonaws.com/oss_compliance.zip \
+   && unzip ${HOME_DIR}/oss_compliance.zip -d ${HOME_DIR}/ \
+   && cp ${HOME_DIR}/oss_compliance/test/testOSSCompliance /usr/local/bin/testOSSCompliance \
+   && chmod +x /usr/local/bin/testOSSCompliance \
+   && chmod +x ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh \
+   && ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh ${HOME_DIR} ${PYTHON} \
+   && rm -rf ${HOME_DIR}/oss_compliance*
+
+RUN curl https://aws-dlc-licenses.s3.amazonaws.com/tensorflow-2.5/license.txt -o /license.txt
```
[the remark this answers is not in the document]
Comment (tejaschumbalkar) [reply habana/tensorflow/training/docker/2.7/py3/synapseai1.1.1/Dockerfile.hpu:131]: @waytrue17 Let use the TF2.7 license `https://aws-dlc-licenses.s3.amazonaws.com/tensorflow-2.7/license.txt`

File: habana/pytorch/training/docker/1.9/py3/synapseai1.1.1/Dockerfile.hpu (hunk 7 above)
Comment (omrialmog) [reply habana/pytorch/training/docker/1.9/py3/synapseai1.1.1/Dockerfile.hpu:58]: @tejaschumbalkar  this can be removed. Thanks

File: habana/tensorflow/training/docker/2.7/py3/synapseai1.1.1/Dockerfile.hpu (hunk 8 above)
[the remark this answers is not in the document]
Comment (waytrue17) [reply habana/tensorflow/training/docker/2.7/py3/synapseai1.1.1/Dockerfile.hpu]: Updated license files.
